### PR TITLE
NLOpt Optimizers

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

File: pypesto/optimize/optimizer.py [hunk 1]
```diff
@@ -630,3 +636,162 @@ def minimize(
...
+        elif 'maxiter' in options:
+            options['maxeval'] = options.pop('maxiter')
+        if local_options is None:
+            local_options = {}
+        self.options = options
+        self.local_options = local_options
+        if nlopt is None:
+            raise ImportError(
+                "This optimizer requires an installation of NLopt. You can "
+                "install NLopt via pip install nlopt.")
```
Comment (LeonardSchmiester) [inline pypesto/optimize/optimizer.py:681]: Is checked twice, here and in `minimize()`.

**Review by LeonardSchmiester — APPROVED** [review]
Looks good to me :+1:

File: pypesto/optimize/optimizer.py [hunk 2]
```diff
@@ -630,3 +636,162 @@ def minimize(
...
+        needs_local_method = [
+            nlopt.G_MLSL, nlopt.G_MLSL_LDS, nlopt.GD_MLSL, nlopt.GD_MLSL_LDS,
+            nlopt.AUGLAG, nlopt.AUGLAG_EQ
+        ]
+
+        if local_method is None and method in needs_local_method:
+            local_method = nlopt.LD_LBFGS
+
+        if local_method is not None and method not in needs_local_method:
+            raise ValueError(f'Method "{method}" does not allow a local '
```
Comment (jvanhoefer) [inline pypesto/optimize/optimizer.py:695]: Shall we rather raise a warning, that the local method is unused and continue without it?

**Review by jvanhoefer — APPROVED** [review]
👍

File: pypesto/optimize/optimizer.py (hunk 2 above)
Comment (FFroehlich) [reply pypesto/optimize/optimizer.py:695]: I see no good reason why a user would want that and I generally like working with sanitized input.

File: pypesto/optimize/optimizer.py (hunk 1 above)
Comment (FFroehlich) [reply pypesto/optimize/optimizer.py:681]: True, if we check in `__init__`, it should be available in `minimize`.

File: test/test_optimize.py [hunk 3]
```diff
@@ -28,6 +29,20 @@ def mode(request):
     ('dlib', ''),
     ('pyswarm', ''),
     ('cmaes', ''),
+    *[('nlopt', method) for method in [
```
Comment (yannikschaelte) [inline test/test_optimize.py:32]: wow, testing all :laughing: but yes, makes sense.

**Review by yannikschaelte — COMMENTED** [review]
I was anyway too late, but looks good! :+1:

File: test/test_optimize.py (hunk 3 above)
Comment (FFroehlich) [reply test/test_optimize.py:32]: with the rosen objective these finish in a matter of a couple of seconds, makes me more confident they are not completely terrible :D

**Pull request merged**